pkg/expression/expression.go

```diff
@@ -1037,7 +1037,7 @@ func ColumnInfos2ColumnsAndNames(ctx sessionctx.Context, dbName, tblName model.C
 			if err != nil {
 				return nil, nil, errors.Trace(err)
 			}
-			e, err := RewriteAstExpr(ctx, expr, mockSchema, names, true)
+			e, err := BuildExprWithAst(ctx, expr, WithInputSchemaAndNames(mockSchema, names), WithSourceTable(tblInfo), WithAllowCastArray(true))
```

Comment:
Only this line changes some logic. We use `BuildExprWithAst` to replace `RewriteAstExpr` because `BuildExprWithAst` is does not require full `session.Context` and it only requires `BuildExprWithAst` to build general expressions.